KnativeDeployer environment check is breaking the build

### Describe the bug

We have apps that use Knative client and now with `kubernetes-client` dependency has been migrated to 6.1.1, the builds are broken. It might be down to changes descried in migration guide https://github.com/fabric8io/kubernetes-client/blob/master/doc/MIGRATION-v6.md#adapt-changes

### Expected behavior

Builds just fine.

### Actual behavior

Exception is thrown:
```
...
12:56:59,665 INFO  mvn: Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
12:56:59,665 INFO  mvn: 	[error]: Build step io.quarkus.kubernetes.deployment.KnativeDeployer#checkEnvironment threw an exception: java.lang.IllegalStateException: Knative was requested as a deployment, but the target cluster is not a Knative cluster!
12:56:59,665 INFO  mvn: 	at io.quarkus.kubernetes.deployment.KnativeDeployer.lambda$checkEnvironment$0(KnativeDeployer.java:29)
12:56:59,665 INFO  mvn: 	at java.base/java.util.Optional.ifPresent(Optional.java:178)
12:56:59,665 INFO  mvn: 	at io.quarkus.kubernetes.deployment.KnativeDeployer.checkEnvironment(KnativeDeployer.java:21)
12:56:59,665 INFO  mvn: 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
12:56:59,665 INFO  mvn: 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
12:56:59,666 INFO  mvn: 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
12:56:59,666 INFO  mvn: 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
12:56:59,666 INFO  mvn: 	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)
12:56:59,666 INFO  mvn: 	at io.quarkus.builder.BuildContext.run(BuildContext.java:281)
12:56:59,666 INFO  mvn: 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
12:56:59,666 INFO  mvn: 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
12:56:59,666 INFO  mvn: 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
12:56:59,666 INFO  mvn: 	at java.base/java.lang.Thread.run(Thread.java:833)
12:56:59,666 INFO  mvn: 	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
12:56:59,667 INFO  mvn: 
12:56:59,667 INFO  mvn:     at io.quarkus.builder.Execution.run (Execution.java:123)
12:56:59,667 INFO  mvn:     at io.quarkus.builder.BuildExecutionBuilder.execute (BuildExecutionBuilder.java:79)
...
```

### How to Reproduce?

Reproducer:

Steps to reproduce the behavior:
0. Have OC stuff ready
1. `git clone https://github.com/quarkus-qe/quarkus-test-suite.git`
2. `cd http/http-minimum`
3. `mvn clean verify -Dit.test=ServerlessExtensionOpenShiftHttpMinimumIT -Dopenshift -Dserverless -Doperator-scenarios`

I can provide standalone reproducer if required.

### Output of `uname -a` or `ver`

Linux fedora 5.19.6-200.fc36.x86_64 #1 SMP PREEMPT_DYNAMIC Wed Aug 31 17:58:15 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk version "17.0.3" 2022-04-19

### GraalVM version (if different from Java)

GraalVM CE 22.1.0

### Quarkus version or git rev

999-SNAPSHOT

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.6

### Additional information

_No response_

--- a/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KnativeDeployer.java
+++ b/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KnativeDeployer.java
@@ -5,7 +5,6 @@ import static io.quarkus.kubernetes.deployment.Constants.KNATIVE;
 import java.util.List;
 import java.util.Optional;
 
-import io.fabric8.knative.client.KnativeClient;
 import io.quarkus.deployment.annotations.BuildProducer;
 import io.quarkus.deployment.annotations.BuildStep;
 import io.quarkus.kubernetes.client.spi.KubernetesClientBuildItem;
@@ -23,7 +22,8 @@ public class KnativeDeployer {
                 return;
             }
             if (target.getEntry().getName().equals(KNATIVE)) {
-                if (client.getClient().isAdaptable(KnativeClient.class)) {
+                // use 'isSupported' once https://github.com/fabric8io/kubernetes-client/issues/4447 is resolved
+                if (client.getClient().hasApiGroup("knative.dev", false)) {
                     deploymentCluster.produce(new KubernetesDeploymentClusterBuildItem(KNATIVE));
                 } else {
                     throw new IllegalStateException(

--- a/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/OpenshiftDeployer.java
+++ b/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/OpenshiftDeployer.java
@@ -24,11 +24,16 @@ public class OpenshiftDeployer {
                 return;
             }
             if (target.getEntry().getName().equals(OPENSHIFT)) {
-                if (client.getClient().isAdaptable(OpenShiftClient.class)) {
-                    deploymentCluster.produce(new KubernetesDeploymentClusterBuildItem(OPENSHIFT));
-                } else {
-                    throw new IllegalStateException(
-                            "Openshift was requested as a deployment, but the target cluster is not an Openshift cluster!");
+                try (var openShiftClient = client.getClient().adapt(OpenShiftClient.class)) {
+                    if (openShiftClient.isSupported()) {
+                        deploymentCluster.produce(new KubernetesDeploymentClusterBuildItem(OPENSHIFT));
+                    } else {
+                        throw new IllegalStateException(
+                                "Openshift was requested as a deployment, but the target cluster is not an Openshift cluster!");
+                    }
+                } catch (Exception e) {
+                    throw new RuntimeException(
+                            "Failed to configure OpenShift. Make sure you have the Quarkus OpenShift extension.", e);
                 }
             }
         });

